### PR TITLE
[DynamicCompilation] Partition @convention(tensorflow) functions in both graph mode and eager mode

### PR DESCRIPTION
**Pull request by rxwei**
This patch makes eager mode also run partitioning for `@convention(tensorflow)` functions, because their graph functions must be statically emitted. For now, "eager mode" refers to a combination of `-O` and `-Xllvm -tf-dynamic-compilation`.

### CODE REVIEW

File: test/TensorFlow/accelerator_only_functions.swift [hunk 1]
```diff
@@ -0,0 +1,285 @@
+// RUN: %target-swift-frontend -Xllvm -tf-dynamic-compilation -Xllvm -tf-dump-graph -Xllvm -tf-module-level-graph=false -O -emit-sil %s | %FileCheck %s
+
+import TensorFlow
+
+@TensorFlowGraph
+public func add(x: Tensor<Float>, y: Tensor<Float>) -> Tensor<Float> {
+  return x + y
+}
+
+// CHECK-LABEL: --- TFPartition GraphDef Proto:
```
Comment (mhong) [inline test/TensorFlow/accelerator_only_functions.swift]: how about just keeping the first CHECK-LABEL, and omit the rest below?

**Comment by mhong** [comment]
Nice! To plumb through IRGen for executing a tensorflow-convention function, we'll probably need to pass a mapping from a SIL func name to graph function name (and also the global string value storing the serialized GraphDef / FunctionDef) to IRGen, so that IRGen can call TF C APIs to add that FunctionDef to TFE_Context, and create & execute a single TFE_Op based on the graph func name.
 
We can discuss later who/when to do this later if you plan to tackle other work first.

**Review by mhong — APPROVED** [review]

**Comment by mhong** [comment]
@marcrasi as FYI

**Comment by rxwei** [comment]
> To plumb through IRGen for executing a tensorflow-convention function, we'll probably need to pass a mapping from a SIL func name to graph function name (and also the global string value storing the serialized GraphDef / FunctionDef) to IRGen, so that IRGen can call TF C APIs to add that FunctionDef to TFE_Context, and create & execute a single TFE_Op based on the graph func name.

Yeah, it can be pretty simple actually. 

Given a call to a `@convention(tensorflow) (T...) -> U` function:
```swift
let f: @convention(tensorflow) (Tensor<Float>, Tensor<Float>) -> Tensor<Float>
//   ^ the graph func name is "whatever.tf_only"
let x = Tensor<Float>(0)
_ = f(x)
```

There are two option here:
1. SILGen `f(x)` to `graph_op "PartitionedCall"(%x)`, which is a function application that takes a function attribute.
2. SILGen `f(x)` to `graph_op "whatever.tf_only"(%x)` because the function can be used as an op.

@alextp suggested that `PartitionedCall` may be better because it supports multiple devices.

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
I can help out more on GPE stuff in the end of the week. Feel free to implement whatever's next if you get to it first!

**Comment by rxwei** [comment]
GPU tests are flakes. Merging.

**Pull request merged**